src/btree/bt_sync_obsolete.c

```diff
@@ -475,7 +476,7 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
     WT_RET(__wt_config_getones(session, config, "checkpoint", &cval));
     __wt_config_subinit(session, &ckptconf, &cval);
     while ((ret = __wt_config_next(&ckptconf, &key, &cval)) == 0) {
-        ret = __wt_config_subgets(session, &cval, "newest_stop_durable_ts", &value);
+        ret = __wt_config_subgets(session, &cval, "newest_page_stop_durable_ts", &value);
```

Comment:
How it is backward compatible if this is changed in checkpoint?

Reply:
There is no problem with backward compatibility, here we just ignore this file until the next checkpoint occurs. When the checkpoint happens, all the data is written in the newer format and it works fine.

The only problem with the current code is downgrading to an older version using an unclean shutdown.

Reply:
> The only problem with the current code is downgrading to an older version using an unclean shutdown.

Yes, this is what I'm worried about. We cannot introduce this to 8.0 as we have a plan to be able to directly upgrade from 5.0 to 8.0.

Reply:
How about writing the old and new format both for one major version and removing it later? It an extra data for the checkpoint, other than it is not going to cause any problem.

Reply:
At the checkpoint level, writing both I think solves the issue. But still do we know will it still wort that if the database have pages with timestamps of different meanings? Also we don't know which page has what types of timestamps.

Reply:
Writing both variable names for the checkpoint has solved the upgrade/downgrade problem. 

> But still do we know will it work that if the database have pages with timestamps of different meanings? Also we don't know which page has what types of timestamps.

We are not changing how we store the timestamp information on the page/cell. It is jut the value that is changed, so there is no problem.